sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java

```diff
@@ -889,6 +904,66 @@ public void changeFeedQueryCompleteAfterAvailableNow(
         }
     }
 
+    // These tests cover a gap that was identified in querying changefeed using hpk
+    @Test(groups = { "emulator" }, dataProvider = "queryChangeFeedWithHierarchicalPartitionKeyDataProvider", timeOut = TIMEOUT)
+    public void queryChangeFeedWithHierarchicalPartitionKey(String pk1, String pk2, int pk3) {
+        String testContainerId = UUID.randomUUID().toString();
+
+        try {
+            PartitionKeyDefinition partitionKeyDefinition = new PartitionKeyDefinition();
+            partitionKeyDefinition.setKind(PartitionKind.MULTI_HASH);
+            partitionKeyDefinition.setVersion(PartitionKeyDefinitionVersion.V2);
+            ArrayList<String> paths = new ArrayList<>();
+            paths.add("/city");
+            paths.add("/zipcode");
+            paths.add("/areaCode"); // expecting int value type
+            partitionKeyDefinition.setPaths(paths);
+
+            PartitionKeyBuilder partialPartitionKeyBuilder = new PartitionKeyBuilder();
+            if (pk1 != null) {
+                partialPartitionKeyBuilder.add(pk1);
+            }
+            if (pk2 != null) {
+                partialPartitionKeyBuilder.add(pk2);
+            }
+            if (pk3 != -1) {
+                partialPartitionKeyBuilder.add(pk3);
+            }
+
+            PartitionKey partialPartitionKey = partialPartitionKeyBuilder.build();
+            CosmosChangeFeedRequestOptions changeFeedRequestOptions = CosmosChangeFeedRequestOptions.createForProcessingFromBeginning(
+                FeedRange.forLogicalPartition(partialPartitionKey)
+            );
+
+            CosmosContainerProperties containerProperties = new CosmosContainerProperties(testContainerId, partitionKeyDefinition);
+            CosmosAsyncContainer testContainer =
+                createCollection(
+                    this.createdAsyncDatabase,
+                    containerProperties,
+                    new CosmosContainerRequestOptions(),
+                    400);
+
+            CosmosMultiHashTest.CityItem cityItem = new CosmosMultiHashTest.CityItem(UUID.randomUUID().toString(), "Redmond", "98052", 1);
+            testContainer.createItem(cityItem).block();
+
+            var responseIterator = testContainer
```

Comment:
var is not part of java 8 so wouldn't be backward compatible and I think that it is preferred to use explicit types.

Reply:
This shoudl actually fail in some CI pipelines - because they would use Java 8

Reply:
fixed